lnutil/dlclib.go

```diff
@@ -0,0 +1,481 @@
+	}
+
+	theirInputsLen := uint64(len(self.TheirFundingInputs))
+	wire.WriteVarInt(&buf, 0, theirInputsLen)
+
+	for i := 0; i < len(self.TheirFundingInputs); i++ {
+		opArr := OutPointToBytes(self.TheirFundingInputs[i].Outpoint)
+		buf.Write(opArr[:])
+		wire.WriteVarInt(&buf, 0, uint64(self.TheirFundingInputs[i].Value))
+	}
+
+	divisionLen := uint64(len(self.Division))
+	wire.WriteVarInt(&buf, 0, divisionLen)
+
+	for i := 0; i < len(self.Division); i++ {
+		wire.WriteVarInt(&buf, 0, uint64(self.Division[i].OracleValue))
+		wire.WriteVarInt(&buf, 0, uint64(self.Division[i].ValueOurs))
+	}
+
+	theirSigLen := uint64(len(self.TheirSettlementSignatures))
+	wire.WriteVarInt(&buf, 0, theirSigLen)
+
+	for i := 0; i < len(self.TheirSettlementSignatures); i++ {
+		wire.WriteVarInt(&buf, 0, uint64(self.TheirSettlementSignatures[i].Outcome))
+		buf.Write(self.TheirSettlementSignatures[i].Signature[:])
+	}
+
+	opArr := OutPointToBytes(self.FundingOutpoint)
+	buf.Write(opArr[:])
+
+	return buf.Bytes()
+}
+
+// GetDivision loops over all division specifications inside the contract and returns the one matching the requested oracle value
+func (c DlcContract) GetDivision(value int64) (*DlcContractDivision, error) {
+	for _, d := range c.Division {
+		if d.OracleValue == value {
+			return &d, nil
+		}
+	}
+
+	return nil, fmt.Errorf("Division not found in contract")
+}
+
+// GetTheirSettlementSignature loops over all stored settlement signatures from the counter party and
+// returns the one matching the requested oracle value
+func (c DlcContract) GetTheirSettlementSignature(value int64) ([64]byte, error) {
+
+	for _, s := range c.TheirSettlementSignatures {
+		if s.Outcome == value {
+			return s.Signature, nil
+		}
+	}
+
+	return [64]byte{}, fmt.Errorf("Signature not found in contract")
+}
+
+// PrintTx prints out a transaction as serialized byte array to StdOut
+func PrintTx(tx *wire.MsgTx) {
+	var buf bytes.Buffer
+	w := bufio.NewWriter(&buf)
+	tx.Serialize(w)
+	w.Flush()
+	fmt.Printf("%x\n", buf.Bytes())
+}
+
+// DlcOutput returns a Txo for a particular value that pays to (PubKeyPeer+PubKeyOracleSig or (OurPubKey and TimeDelay))
+func DlcOutput(pubKeyPeer, pubKeyOracleSig, ourPubKey [33]byte, value int64) *wire.TxOut {
+	scriptBytes := DlcCommitScript(pubKeyPeer, pubKeyOracleSig, ourPubKey, 5)
+	scriptBytes = P2WSHify(scriptBytes)
+
+	return wire.NewTxOut(value, scriptBytes)
+}
+
+// DLC Commit script makes a script that pays to (PubKeyPeer+PubKeyOracleSig or (OurPubKey and TimeDelay))
+// We send this over (signed) to the other side. If they publish the TX with the correct script they can use
+// The oracle's signature and their own private key to claim the funds from the output. However,
+// If they send the wrong one, they won't be able to claim the funds - and we can claim them once the
+// Time delay has passed.
+func DlcCommitScript(pubKeyPeer, pubKeyOracleSig, ourPubKey [33]byte, delay uint16) []byte {
```

Comment:
This is almost the same as CommitScript() in lnlib with, so clearer to call CombinePubs() and then CommitScript() which would make this function ~2 lines